tree/tree.go

```diff
@@ -0,0 +1,76 @@
+package tree
+
+import "fmt"
+
+// Tree holds the tree data
+type Tree struct {
+	rootNode Node
+}
+
+// NewTree returns a new instance of the Tree
+func NewTree() *Tree {
+	return &Tree{
+		rootNode: NewRootNode(),
+	}
+}
+
+func (t *Tree) Root() []byte {
+	return t.rootNode.GetHash()
+}
+
+func (t *Tree) Get(key []byte) ([]byte, bool) {
+	node := t.findNode(FromBytes(key), t.rootNode)
+	if _, ok := node.(*EmptyNode); ok {
+		return nil, false
+	}
+	return node.Value(), true
+}
+
+// Put
+// - When inserting in a EmptyNode replace it with a LeafNode
+// - When inserting in a BranchNode
+func (t *Tree) Put(key []byte, value []byte) {
```

Comment:
The pattern we have elsewhere (namely the database.Database interface) is that Put returns  and `error`

Reply:
Fixed :)